lib/ClangImporter/ImportType.cpp

```diff
@@ -1771,15 +1771,29 @@ adjustResultTypeForThrowingFunction(ForeignErrorConvention::Info errorInfo,
   switch (errorInfo.TheKind) {
   case ForeignErrorConvention::ZeroResult:
   case ForeignErrorConvention::NonZeroResult:
+    // Check for a bad override.
+    if (resultTy->isVoid())
+      return Type();
     return TupleType::getEmpty(resultTy->getASTContext());
 
   case ForeignErrorConvention::NilResult:
-    resultTy = resultTy->getAnyOptionalObjectType();
-    assert(resultTy &&
-           "result type of NilResult convention was not imported as optional");
-    return resultTy;
+    if (Type unwrappedTy = resultTy->getAnyOptionalObjectType())
+      return unwrappedTy;
+    // It's possible an Objective-C method overrides the base method to never
+    // fail, and marks the method _Nonnull to indicate that. Swift can't
+    // represent that, but it shouldn't fall over either.
+    if (resultTy->isAnyClassReferenceType()
+        || resultTy->is<AnyFunctionType>()
+        || resultTy->getAnyPointerElementType())
```

Comment:
This is an odd set of conditions that excludes bridged types. `adjustResultTypeForThrowingFunction` 's caller allows bridging, so an `NSString * _Nonnull` result type would fail to import (because `String` doesn't meet any of those conditions) while an `NSObject * _Nonnull` result type would smooth over the issue in the ObjC headers. 

If we're not going to rigorously check for consistency, then an `isVoid` check probably suffices here as in the other places.